### PR TITLE
Quick updates to the RFS design doc, assuming a later pass

### PR DESCRIPTION
**Pull request by chelma**
### Description
Performed a quick update to the RFS Design doc to clear out most of the outdated stuff.  I assumed a follow-up pass would be made by @mikaylathompson in her update to he work coordination behavior (see: https://opensearch.atlassian.net/browse/MIGRATIONS-2126).

### Issues Resolved
* https://opensearch.atlassian.net/browse/MIGRATIONS-2129

### Testing
N/A

### Check List
- [ ] New functionality includes testing
  - [ ] All tests pass, including unit test, integration test and doctest
- [X] New functionality has been documented
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: RFS/docs/DESIGN.md [hunk 1]
```diff
@@ -77,20 +76,17 @@ Below is an example for the structure of an Elasticsearch 7.10 snapshot, along w
 8. **Snapshot Metadata File**: SMILE encoded; contains things like whether the snapshot succeeded, the number of shards, how many shards succeeded, the ES/OS version, the indices in the snapshot, etc
 
 ## Ultra-High Level Design
-The responsibility of performing an RFS operation is split into two groups of actors - the RFS Workers, and RFS Scaler (see Figure 1, below).
+The responsibility of performing an RFS operation is performed by a group of one or more the RFS Workers (see Figure 1, below).
```
Comment (gregschohn) [inline RFS/docs/DESIGN.md:79]: ...one or more ~~the~~ RFS...

File: RFS/docs/DESIGN.md [hunk 2]
```diff
@@ -142,13 +134,13 @@ The unit of work for an RFS Worker migrating Elasticsearch Documents is an Elast
...
 
-In this section, we describe in a high-level, narrative manner how the RFS Worker operates. Detailed state machine diagrams that outline the behavior more explicitly are below as well (see [Appendix: RFS Worker State Machine](#appendix-rfs-worker-state-machine)).  The state machine diagram is intended to be the source of truth, so favor its representation over the narrative description below if conflicts exist.
+In this section, we describe in a high-level, narrative manner how the RFS Worker operates.
 
 ### RFS Worker threads
 
 The RFS Worker’s running process has at least two running threads:
 
-* Main Thread - performs the work of moving the data/metadata from the source cluster to the target cluster; starts the Metrics and Healthcheck Threads
+* Main Thread - performs the work of moving the data from the source cluster to the target cluster; starts the Metrics and Healthcheck Threads
```
Comment (gregschohn) [inline RFS/docs/DESIGN.md:143]: There likely is a thread for metrics, but do you care to mention it?  There's also more than 1 thread responsible for moving data to the target now that we're using more work fanouts from reactor.

File: RFS/docs/DESIGN.md [hunk 3]
```diff
@@ -222,25 +214,18 @@ If no Entry is returned, we know that this sub-phase is complete and attempt to
...
-* (A4) - The creation of Elasticsearch Templates is atomic
-* (A5) - The creation of Elasticsearch Indices is atomic
-* (A6) - Re-doing portions of the overall RFS operation is fine, as long as every portion is completed at least once
-* (A7) - Elasticsearch Templates must be migrated in the order: first Legacy, then Composite, then Index
-* (A8) - All Elasticsearch Templates must be migrated before any Elasticsearch Indices can be migrated
-* (A9) - Elasticsearch Indices can be migrated in parallel without ordering concerns
-* (A10) - Elasticsearch Shards can only be migrated after their Elasticsearch Index has been migrated
-* (A11) - Elasticsearch Shards can be migrated in parallel without ordering concerns
+* (A3) - Re-doing portions of the overall RFS operation is fine, as long as every portion is completed at least once
+* (A4) - Elasticsearch Shards can only be migrated after their Elasticsearch Index has been migrated
```
Comment (gregschohn) [inline RFS/docs/DESIGN.md:227]: Given the discussions about defaults and manually setting metadata, this could also be "migrated or created"

**Review by gregschohn — APPROVED** [review]
Minor comments/typos, but LGTM

File: RFS/docs/DESIGN.md (hunk 2 above)
Comment (chelma) [reply RFS/docs/DESIGN.md:143]: Just missed that reference to the Metrics thread; thanks for catching.

IIRC from convos w/ @mikaylathompson, we're going to be move from multiple threads for reindexing to a single thread in order for the checkpointing to work as intended.  We'll just need to make sure that we update the doc here for the final behavior.

**Pull request merged**